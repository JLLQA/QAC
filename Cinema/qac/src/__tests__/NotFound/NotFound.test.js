import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Notfound from '../../Components/NotFound/NotFound';
import renderer from 'react-test-renderer';

describe('Test for 404 page', () => {
test('Check the 404 heading is present', () => {
    render(<Notfound />);
    const linkElement = screen.getByText("~~404~~");
    expect(linkElement).toBeInTheDocument();
});

test('Check the body text to be present', () => {
    render(<Notfound />);
    const linkElement = screen.getByText("You silly goose");
    expect(linkElement).toBeInTheDocument();
});

test("It matches the snapshot", () => {
    const tree = renderer.create(<Notfound />).toJSON();
    expect(tree).toMatchSnapshot();
});
})