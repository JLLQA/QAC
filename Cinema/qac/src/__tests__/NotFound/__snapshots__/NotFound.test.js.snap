// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for 404 page It matches the snapshot 1`] = `
<div
  className="container-fluid"
  id="dropped-box"
>
  <div
    align="center"
    className="container"
  >
    <div
      className="container"
    >
      <h1>
        ~~404~~
      </h1>
    </div>
    <br />
    <div
      className="container"
    >
      <h3>
        You silly goose
      </h3>
      <h3>
        We're sorry but that page can't be found
      </h3>
    </div>
    <br />
    <div
      className="container"
    >
      <img
        alt="basicplan"
        id="plan"
        src="https://ichef.bbci.co.uk/news/976/cpsprodpb/12B60/production/_109004667_02untitledgoosegamescreen3840x2160.png"
      />
    </div>
    <br />
    <div
      className="container"
    >
      <h3>
        Don't worry we're working on it
      </h3>
    </div>
  </div>
</div>
`;

exports[`test for 404 page It matches the snapshot 1`] = `
<div
  className="container-fluid"
  id="dropped-box"
>
  <div
    align="center"
    className="container"
  >
    <div
      className="container"
    >
      <h1>
        ~~404~~
      </h1>
    </div>
    <br />
    <div
      className="container"
    >
      <h3>
        You silly goose
      </h3>
      <h3>
        We're sorry but that page can't be found
      </h3>
    </div>
    <br />
    <div
      className="container"
    >
      <img
        alt="basicplan"
        id="plan"
        src="https://ichef.bbci.co.uk/news/976/cpsprodpb/12B60/production/_109004667_02untitledgoosegamescreen3840x2160.png"
      />
    </div>
    <br />
    <div
      className="container"
    >
      <h3>
        Don't worry we're working on it
      </h3>
    </div>
  </div>
</div>
`;

exports[`test for contact page It matches the snapshot 1`] = `
<div
  className="container-fluid"
  id="dropped-box"
>
  <div
    align="center"
    className="container"
  >
    <div
      className="container"
    >
      <h1>
        ~~404~~
      </h1>
    </div>
    <br />
    <div
      className="container"
    >
      <h3>
        You silly goose
      </h3>
      <h3>
        We're sorry but that page can't be found
      </h3>
    </div>
    <br />
    <div
      className="container"
    >
      <img
        alt="basicplan"
        id="plan"
        src="https://ichef.bbci.co.uk/news/976/cpsprodpb/12B60/production/_109004667_02untitledgoosegamescreen3840x2160.png"
      />
    </div>
    <br />
    <div
      className="container"
    >
      <h3>
        Don't worry we're working on it
      </h3>
    </div>
  </div>
</div>
`;
